src/main/java/org/mozilla/msrp/platform/vertical/video/VideoService.kt

```diff
@@ -20,38 +20,32 @@ class VideoService @Inject constructor(
     private val CACHE_THRESHOLD = 30 * 24 * 60 * 60 * 1000L // a month
 
     fun fromCache(query: String, order: String, maxResult: Long): List<VideoItem> {
-        val cacheKey = query
-
-        val content: String? = videoCacheRepository.get(query)
-        // if not in cache
-        if (content == null) {
-            VerticalMetrics.event(VerticalMetrics.EVENT_CACHE_MISSED, cacheKey)
-            val data = loadData(query, order, maxResult)
-            videoCacheRepository.set(cacheKey, mapper.writeValueAsString(data))
```

Comment:
there's an error here.
When there's no result from `loadData(...)`, the empty result will still be written to the cache.